transport/src/main/java/io/zeebe/transport/ServerOutput.java

```diff
@@ -7,22 +7,17 @@
  */
 package io.zeebe.transport;
 
-import io.zeebe.util.buffer.BufferWriter;
+import io.zeebe.transport.impl.ServerResponseImpl;
 
 public interface ServerOutput {
   /**
-   * Sends a message according to the single message protocol.
+   * Sends the given response. The corresponding partition and request id is extracted from the
+   * response object.
    *
-   * <p>Returns false if the message cannot be currently written due to exhausted capacity. Throws
-   * an exception if the request is not sendable at all (e.g. buffer writer throws exception).
-   */
-  boolean sendMessage(int streamId, BufferWriter writer);
-
-  /**
-   * Sends a response according to the request response protocol.
+   * <p>This method should decouple the the request handling, such that response sending can be done
+   * later asynchronously.
    *
-   * <p>Returns null if the response cannot be currently written due to exhausted capacity. Throws
-   * an exception if the response is not sendable at all (e.g. buffer writer throws exception).
+   * @param response the response which should be send
    */
-  boolean sendResponse(ServerResponse response);
+  void sendResponse(ServerResponseImpl response);
```

Comment:
I think in the interface we should use the interface `ServerResponse` instead of the implementation `ServerResponseImpl`, right?

Reply:
Oh yes good catch thanks you're right